Games/Minesweeper/Field.cpp

```diff
@@ -271,15 +272,40 @@ void Field::reset()
     set_updates_enabled(true);
 }
 
+void Field::flood_mark(Square& square)
+{
+    if (square.is_swept)
+        return;
+    if (square.has_flag)
+        return;
+    if (square.is_considering)
+        return;
+    update();
+    square.is_swept = true;
+    square.button->set_visible(false);
+    square.label->set_visible(true);
+
+    --m_unswept_empties;
+
+    if (!m_unswept_empties)
+        win();
+}
+
 void Field::flood_fill(Square& square)
 {
-    on_square_clicked(square);
-    square.for_each_neighbor([this](auto& neighbor) {
-        if (!neighbor.is_swept && !neighbor.has_mine && neighbor.number == 0)
-            flood_fill(neighbor);
-        if (!neighbor.has_mine && neighbor.number)
-            on_square_clicked(neighbor);
-    });
+    DoublyLinkedList<Square*> queue;
```

Comment:
I think it would be more appropriate to use a `Queue` here, the variable name is kind of a giveaway ;)

You can find an example of a flood fill using `Queue` in `Applications/PaintBrush/BucketTool.cpp`

`Queue` uses a list of `Vector`s internally so it's a lot more space efficient than the scattered `DoublyLinkedList`.